### PR TITLE
qt6: Add macdeployqt patch

### PR DESCRIPTION
**Pull request by jonaski**
#### Description

Fixes deploying application libraries with .so extension such as glib gio modules, this also applies to gstreamer since macports currently uses an older version of gstreamer which uses .so extensions for plugins. See https://codereview.qt-project.org/c/qt/qtbase/+/507393

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Review by chrstphrchvz — CHANGES_REQUESTED** [review]
The revision of qtbase should be incremented, correct?

File: aqua/qt6/files/patch-qtbase-macdeployqt.diff
Comment (chrstphrchvz) [inline aqua/qt6/files/patch-qtbase-macdeployqt.diff:21]: `patch -p0` is used by default, so I think the prefix needs to be removed (see other qt6 patches).

```suggestion
--- src/tools/macdeployqt/shared/shared.cpp
+++ src/tools/macdeployqt/shared/shared.cpp
```

**Review by chrstphrchvz — CHANGES_REQUESTED** [review]
See item “Wrap body text at 72 characters” on https://trac.macports.org/wiki/CommitMessages#guidelines

I have not tested the patch, but I would be okay with it especially since it is a backport. I am not a committer and cannot start the CI for this PR.

**Comment by pmetzger** [comment]
I note that the CI is still broken.

**Comment by pmetzger** [comment]
@chrstphrchvz Please let us know when you think this is in good shape.

**Comment by chrstphrchvz** [comment]
> I note that the CI is still broken.

CI was not broken, it just doesn’t run automatically for someone who hasn’t already contributed to a repository.

**Comment by chrstphrchvz** [comment]
> > I note that the CI is still broken.
> 
> CI was not broken, it just doesn’t run automatically for someone who hasn’t already contributed to a repository.

Never mind, you meant that the CI was still failing.

**Review by chrstphrchvz — APPROVED** [review]
The CI was able to build qt6-qtbase, which I believe is the only port affected by this PR. Any CI failures are the same as those in https://github.com/macports/macports-ports/pull/20562#issuecomment-1741568975. I think this is ready to merge.

**Comment by chrstphrchvz** [comment]
I meant to check for this again before clicking “approve” but this still applies:

> See item “Wrap body text at 72 characters” on https://trac.macports.org/wiki/CommitMessages#guidelines

**Pull request merged**
